config/auth_azure_cli.go

```diff
@@ -128,3 +136,30 @@ func (ts *azureCliTokenSource) Token() (*oauth2.Token, error) {
 		Expiry:       expiresOn,
 	}).WithExtra(extra), nil
 }
+
+func (ts *azureCliTokenSource) getTokenBytes() ([]byte, error) {
+	subscription := ts.getSubscription()
+	args := []string{"account", "get-access-token", "--resource",
+		ts.resource, "--output", "json"}
+	if subscription != "" {
+		extendedArgs := make([]string, len(args))
+		copy(extendedArgs, args)
+		extendedArgs = append(extendedArgs, "--subscription", subscription)
+		// This will fail if the user has access to the workspace, but not to the subscription
+		// itself.
+		// In such case, we fall back to not using the subscription.
+		result, err := exec.Command("az", extendedArgs...).Output()
+		if err == nil {
+			return result, nil
+		}
+		logger.Warnf(context.Background(), "Failed to get token for subscription. Using resource only token.")
+	}
+	result, err := exec.Command("az", args...).Output()
+	if ee, ok := err.(*exec.ExitError); ok {
+		return nil, fmt.Errorf("cannot get access token: %s", string(ee.Stderr))
+	}
+	if err != nil {
+		return nil, fmt.Errorf("cannot get access token: %v", err)
```

Comment:
We should use `%w` to wrap errors, unless for some reason we don't want to expose the underlying error?